crates/rethnet_evm/src/state/layered/changes.rs

```diff
@@ -344,6 +344,13 @@ impl LayeredChanges<RethnetLayer> {
     pub fn remove_code(&mut self, code_hash: &B256) {
         self.last_layer_mut().contracts.remove(code_hash);
     }
+
+    pub fn insert_account(&mut self, address: &Address, account_info: &AccountInfo) {
+        self.account_or_insert_mut(address).info = account_info.clone();
+        if let Some(code) = account_info.code.clone() {
+            self.insert_code(code);
+        }
+    }
```

Comment:
```suggestion
    pub fn insert_account(&mut self, address: &Address, mut account_info: AccountInfo) {
        if let Some(code) = account_info.code.take() {
            self.insert_code(code);
        }

        self.account_or_insert_mut(address).info = account_info;
    }
```

Reply:
1) As we're inserting, we can just explicitly state the intention in the API that we're moving the ownership of the `AccountInfo` instead of requesting a reference.
2) We don't want to store the code in the account itself but store it in a separate data structure. The method to move ownership of a value in an `Option` is by using the `take` method. We do this before assigning the `account_info` to the account